webapp/src/js/controllers/tasks-content.js

```diff
@@ -106,23 +106,25 @@ angular.module('inboxControllers').controller('TasksContentCtrl',
         return $q.resolve(task);
       }
 
-      return DB().get(task.forId)
-        .then(contactDoc => {
-          for (const action of task.actions) {
-            if (!action.content) {
-              action.content = {};
-            }
-
-            if (!action.content.contact) {
-              action.content.contact = contactDoc;
-            }
-          }
+      const setActionsContacts = (task, contact) => {
+        task.actions.forEach(action => {
+          action.content = action.content || {};
+          action.content.contact = action.content.contact || contact;
+        });
+      };
 
+      return DB()
+        .get(task.forId)
+        .catch(err => {
+          if (err.status !== 404) {
+            throw err;
+          }
 
-          return task;
+          $log.info('Failed to hydrate contact information in task action', err);
+          return { _id: task.forId };
         })
-        .catch(err => {
-          $log.error('Failed to hydrate contact information in task action', err);
+        .then(contactDoc => {
+          setActionsContacts(task, contactDoc);
           return task;
```

Comment:
My preferred approach would have been to detect whether a contact is known here: https://github.com/medic/cht-core/blob/master/shared-libs/rules-engine/src/transform-task-emission-to-doc.js#L77 and keep the entire `action.content` when the contact is not known. 
However, there's no straight forward way to have access to that information, and any bigger change would be prone to more bugs or performance issues. 
The only disadvantage with the approach I implemented is that the form will lack all patient information except their id.